api/redhatopenshift/HcpCluster.Management/hcpCluster-models.tsp

```diff
@@ -37,93 +31,49 @@ model HcpOpenShiftClusterResource
 // - deleted when resource group is deleted
 // - allow cascade delete
 // more: https://armwiki.azurewebsites.net/rp_onboarding/tracked_vs_proxy_resources.html
-@parentResource(HcpOpenShiftClusterResource)
-model HcpOpenShiftClusterNodePoolResource
-  is TrackedResource<NodePoolProperties> {
-  /** Name of HCP cluster * */
-  @pattern("^[a-zA-Z][a-zA-Z0-9-]$")
-  @minLength(3)
-  @maxLength(15)
-  @key("nodePoolName")
-  @path
-  @segment("nodePools")
-  name: string;
-
+@parentResource(HcpOpenShiftCluster)
+model NodePool is TrackedResource<NodePoolProperties> {
+  ...ResourceNameParameter<
+    NodePool,
+    NamePattern = "^[a-zA-Z][-a-zA-Z0-9]{1,13}[a-zA-Z0-9]$"
```

Comment:
This is not correct because of the same reasons explained in https://github.com/Azure/ARO-HCP/pull/1510#discussion_r2008032037 but applied to node pool names.

The only difference in this case is the max length which for nodepools name should 15 characters